WORKSPACE

```diff
@@ -210,8 +216,11 @@ node_repositories(
         "https://storage.googleapis.com/public-bazel-artifacts/js/node/v{version}/{filename}",
     ],
     node_version = "16.13.0",
-    package_json = ["//pkg/ui:package.json"],
-    yarn_repositories = {
```

Comment:
This change is causing the generated code check to fail with `"did not parse all needed data from node_repositories call"`. I'll need to patch `pkg/cmd/generate-distdir` to deal with this, give me a second.

Reply:
Try the following patch please and `dev generate bazel`. I haven't tested this, it might need more tweaking, let me know if this doesn't work.

```
diff --git a/pkg/cmd/generate-distdir/main.go b/pkg/cmd/generate-distdir/main.go
index 5b5f5c80e9..4c0b7ec67b 100644
--- a/pkg/cmd/generate-distdir/main.go
+++ b/pkg/cmd/generate-distdir/main.go
@@ -111,7 +111,7 @@ func getShasFromGoDownloadSdkCall(call *syntax.CallExpr, shas map[string]string)
 }
 
 func getShasFromNodeRepositoriesCall(call *syntax.CallExpr, shas map[string]string) error {
-	var nodeURLTmpl, yarnURLTmpl, nodeVersion, yarnVersion, yarnSha, yarnFilename string
+	var nodeURLTmpl, nodeVersion string
 	nodeBaseToHash := make(map[string]string)
 	for _, arg := range call.Args {
 		switch bx := arg.(type) {
@@ -151,7 +151,31 @@ func getShasFromNodeRepositoriesCall(call *syntax.CallExpr, shas map[string]stri
 				if err != nil {
 					return err
 				}
-			} else if kwarg == "yarn_repositories" {
+			}
+		}
+	}
+	if nodeURLTmpl == "" || nodeVersion == "" {
+		return fmt.Errorf("did not parse all needed data from node_repositories call")
+	}
+	for base, sha := range nodeBaseToHash {
+		shas[strings.ReplaceAll(strings.ReplaceAll(nodeURLTmpl, "{version}", nodeVersion), "{filename}", base)] = sha
+	}
+	return nil
+}
+
+func getShasFromYarnRepositoriesCall(call *syntax.CallExpr, shas map[string]string) error {
+	var yarnURLTmpl, yarnVersion, yarnSha, yarnFilename string
+	for _, arg := range call.Args {
+		switch bx := arg.(type) {
+		case *syntax.BinaryExpr:
+			if bx.Op != syntax.EQ {
+				return fmt.Errorf("unexpected binary expression Op %d", bx.Op)
+			}
+			kwarg, err := starlarkutil.ExpectIdent(bx.X)
+			if err != nil {
+				return err
+			}
+			if kwarg == "yarn_releases" {
 				switch d := bx.Y.(type) {
 				case *syntax.DictExpr:
 					if len(d.List) != 1 {
@@ -184,12 +208,9 @@ func getShasFromNodeRepositoriesCall(call *syntax.CallExpr, shas map[string]stri
 			}
 		}
 	}
-	if nodeURLTmpl == "" || yarnURLTmpl == "" || nodeVersion == "" || yarnVersion == "" || yarnSha == "" || yarnFilename == "" {
+	if yarnURLTmpl == "" || yarnVersion == "" || yarnSha == "" || yarnFilename == "" {
 		return fmt.Errorf("did not parse all needed data from node_repositories call")
 	}
-	for base, sha := range nodeBaseToHash {
-		shas[strings.ReplaceAll(strings.ReplaceAll(nodeURLTmpl, "{version}", nodeVersion), "{filename}", base)] = sha
-	}
 	shas[strings.ReplaceAll(strings.ReplaceAll(yarnURLTmpl, "{version}", yarnVersion), "{filename}", yarnFilename)] = yarnSha
 	return nil
 }
@@ -227,6 +248,11 @@ func getShasFromWorkspace(workspace string, shas map[string]string) error {
 						return err
 					}
 				}
+				if fun == "yarn_repositories" {
+					if err := getShasFromYarnRepositoriesCall(x, shas); err != nil {
+						return err
+					}
+				}
 
 			}
 		}

```